gettsim/visualization.py

```diff
@@ -194,17 +195,27 @@ def plot_dag(
     )
     # choose the different options for plotting
 
-    if (len(names) > 10) or (show_labels is False):
-        # For more than 10 nodes or show_labels is False,
-        # the labels are shown when hovering over it.
-        # Otherwise, they are displayed next to the nodes.
+    # When show_lebels = None and number of nodes >10
+    # labels are shown when hovering over it.
+    # Same happens when show_labels is False,
+    # Otherwise, labels are displayed next to the nodes.
 
-        mode = "markers"
-        hover_info = "text"
-    else:
+    if show_labels is None:
```

Comment:
You can combine the conditions more efficiently:

if show_labels or (show_labels is None and len(names) <= 10):
    asdf
else:
    asdf